### DISCUSSION TITLE
What happens to my history when I add or remove partitions?

### DISCUSSION THREAD

**Discussion by clairelin135** [dagster-io/dagster#15462] [Q&A] [answered]
For example, if I have a daily partitions def starting in January 2023, but I change it to start instead in June 2023, what happens to my partitions history?

**Comment by clairelin135** [comment, marked as answer]
When you materialize asset partitions, Dagster stores a materialization event corresponding to the specific partition key. For example, If I materialize the 2023-01-01 partition, Dagster stores an event corresponding to this partition, allowing the UI to display that it has been materialized.

When a partitions def changes to contain more partitions, Dagster reads the existing events, recognizing which partitions have previously been materialized. So in the example above, Dagster can discern whether partitions starting in June 2023 have previously been materialized. **These events are not mutated when the partitions def changes**, so if you then changed your partitions def back (to start in January) your previous history would be displayed.

**Reply by ytoast** [reply]
In my case, I am using StaticPartitions. What if I need to rename my partitions? For example from `aaa` to `title_aaa`. When I do that I will be erasing my previous run history and the UI will not recognise that `title_aaa` has already been run. Is there anyway for me to preserve that history on the UI?

I know you mentioned that the events will not be mutated. Is there a workaround for this?

**Reply by clairelin135** [reply]
Unfortunately it's not currently possible to rename partitions. We do have a feature request for this though: https://github.com/dagster-io/dagster/issues/15444
